### PR TITLE
MGMT-19171: Fix nmpolicy injection into nmstate YAML

### PR DESCRIPTION
**Pull request by linoyaslan**
The current solution doesn't account for situations where a user might select a name that matches one of the YAML keys or even a single character. The following change addresses this issue.

Some edge cases were checked, including:
```
interfaces:
  - name: t
    type: ethernet
    state: up
    ipv4:
      enabled: true
      address:
        - ip: 192.168.101.3
          prefix-length: 24
      dhcp: false
dns-resolver:
    config:
        server:
          - 192.168.101.1
routes:
    config:
    - destination: 0.0.0.0/0
      next-hop-address: 192.168.101.1
      next-hop-interface: t
      table-id: 254
```
```
interfaces:
- name: bond99
  type: bond
  state: up
  ipv4:
    address:
    - ip: 192.0.2.0
      prefix-length: 24
    enabled: true
  link-aggregation:
    mode: balance-rr
    options:
      miimon: '140'
    port:
    - t
    - p
```

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [X] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [X] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [X] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: pkg/staticnetworkconfig/generator.go [hunk 1]
```diff
@@ -137,15 +128,15 @@ func (s *StaticNetworkConfigGenerator) injectNMPolicyCaptures(hostConfig *models
...
-		// VLAN is a special case where the base-iface refers to the logical name rather than the VLAN's name.
-		if val, exists := nameToType[mac.LogicalNicName]; exists && val == "vlan" {
-			re := regexp.MustCompile(fmt.Sprintf("base-iface: %s", interfaceName))
-			modifiedInterfaceConfig = re.ReplaceAllString(interfacesWithCaptures, fmt.Sprintf(`base-iface: "{{ capture.iface%d.interfaces.0.name }}"`, j))
-		} else {
-			re := regexp.MustCompile(interfaceName)
-			modifiedInterfaceConfig = re.ReplaceAllString(interfacesWithCaptures, fmt.Sprintf(`"{{ capture.iface%d.interfaces.0.name }}"`, j))
-
-		}
+		re := regexp.MustCompile(fmt.Sprintf(`(:|-) %s($|\s)`, interfaceName))
```
Comment (danielerez) [inline pkg/staticnetworkconfig/generator.go:131]: Maybe worth adding some of the edge-cases as unit-tests?
Also, a few examples in the PR description, just for context.

Comment (AlonaKaplan) [reply pkg/staticnetworkconfig/generator.go:131]: I think that having multiple spaces in the yaml is valid as well e.g "- name:          eth0". 
Also, what if the interface name is surrounded with quotes e.g "- name: "eth0" or name: 'eth0'
I would unmarshal->marshal the yaml, to normalize it (I believe it will solve both of the issues I mentioned).
And as derez mentioned,  please cover this flow with unit tests (with corner cases + regular flow + the bug the PR is fixing).

**Comment by AlonaKaplan** [comment]
> I think that having multiple spaces in the yaml is valid as well e.g "- name: eth0".
Also, what if the interface name is surrounded with quotes e.g "- name: "eth0" or name: 'eth0'
I would unmarshal->marshal the yaml, to normalize it (I believe it will solve both of the issues I mentioned).
And as derez mentioned, please cover this flow with unit tests (with corner cases + regular flow + the bug the PR is fixing).

Can you please add unit tests to cover the scenarios described in this comment?

**Comment by linoyaslan** [comment]
@AlonaKaplan I’ve already added the unit tests mentioned. Is there anything I missed? -https://github.com/openshift/assisted-service/pull/6948/files#diff-1817322ea76162fd57cba6f3a30bfe7dfa3dc1b6f608ee102f272e40152608c4R678-R692

**Comment by linoyaslan** [comment]
@AlonaKaplan I added the one with multiple spaces that I missed - https://github.com/openshift/assisted-service/pull/6948/commits/d3836a56935b332cfe890efa68f6e4185ecbb79f#diff-1817322ea76162fd57cba6f3a30bfe7dfa3dc1b6f608ee102f272e40152608c4R694
